source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,23 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
```

Comment:
This whole function is low-level C instead of C++11.

Please consider using `std::string::back()` and `std::string::compare(0, uriPattern.length() - 1, uri)`